File: Java/백준/Silver/1748. 수 이어 쓰기 1/수 이어 쓰기 1.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {

        Scanner s = new Scanner(System.in);

        int N = s.nextInt(); //최대 10^9까지
        int max = 0;
        double digits = 0;

        for (int i = 9; i >= 0; i--) {
            if (N - Math.pow(10, i) >= 0) {
                max = i;
                digits = (i + 1) * (N - Math.pow(10, i) + 1);
                break;
            }
        }

        for (int j = max; j > 0; j--) {
            digits += j * (Math.pow(10, j - 1) * 9);
        }

        System.out.print((long) digits);
    }
}

File: C11/백준/Silver/1748. 수 이어 쓰기 1/수 이어 쓰기 1.c
#include <stdio.h>
#include <math.h>

int main(void){
	
	int N, max;
	double digits = 0;
	
	scanf("%d", &N);
	
	for(int i = 9; i >= 0; i--){
		if(N - pow(10, i) >= 0){
			max = i;
			digits = (i + 1) * (N - pow(10, i) + 1);
			break;
		}
	}
	
	for(int j = max; j > 0; j--){
		digits += j * (pow(10, j - 1) * 9);
	}
	
	printf("%ld", (long) digits);
	
	return 0;
}